From 12601373d287f0bf0460b2bc81fda7aa7cee58f2 Mon Sep 17 00:00:00 2001
From: Joshua Chia <joshchia@gmail.com>
Date: Tue, 14 Dec 2021 01:37:41 +0800
Subject: [PATCH] Fixed compilation errors for GHC-9.0.1

Added eta-expansion related to implicit parameters and removed
constraint from Taffy type synonym.
---
 src/System/Taffybar/Context.hs        | 4 ++--
 src/System/Taffybar/Widget/Windows.hs | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/System/Taffybar/Context.hs b/src/System/Taffybar/Context.hs
index 184827d7..ab86cf57 100644
--- a/src/System/Taffybar/Context.hs
+++ b/src/System/Taffybar/Context.hs
@@ -87,7 +87,7 @@
 logC :: MonadIO m => System.Log.Logger.Priority -> String -> m ()
 logC p = liftIO . logIO p
 
-type Taffy m v = MonadIO m => ReaderT Context m v
+type Taffy m v = ReaderT Context m v
 type TaffyIO v = ReaderT Context IO v
 type Listener = Event -> Taffy IO ()
 type SubscriptionList = [(Unique, Listener)]
@@ -226,7 +226,7 @@ buildContext TaffybarConfig
                 }
   _ <- runMaybeT $ MaybeT GI.Gdk.displayGetDefault >>=
               (lift . GI.Gdk.displayGetDefaultScreen) >>=
-              (lift . flip GI.Gdk.afterScreenMonitorsChanged
+              (lift . (\x y -> GI.Gdk.afterScreenMonitorsChanged y x)
                -- XXX: We have to do a force refresh here because there is no
                -- way to reliably move windows, since the window manager can do
                -- whatever it pleases.
diff --git a/src/System/Taffybar/Widget/Windows.hs b/src/System/Taffybar/Widget/Windows.hs
index 9cb43096..1f07e07c 100644
--- a/src/System/Taffybar/Widget/Windows.hs
+++ b/src/System/Taffybar/Widget/Windows.hs
@@ -79,7 +79,7 @@ windowsNew config = do
   subscription <-
     subscribeToPropertyEvents [ewmhActiveWindow, ewmhWMName, ewmhWMClass]
                       activeWindowUpdatedCallback
-  _ <- liftReader (Gtk.onWidgetUnrealize label) (unsubscribe subscription)
+  _ <- liftReader (\x -> Gtk.onWidgetUnrealize label x) (unsubscribe subscription)
 
   context <- ask
 
